### ISSUE TITLE
Decide on a set of priority locales

### ISSUE THREAD

**Issue by davidflanagan** [mdn/sprints#1595]
# User story
As the MDN product owner, I want to designate our most frequently used locales so that we can optimize performance for those locales.

As part of #1587 Ryan and I would like to have a set of 5 to 10 most used locales for which we will pre-render documents so that they can be served quickly. We don't want to have to do this for all 60+ locales that we currently support.

# Acceptance criteria

- The most frequently used (from GA, presumably) or most strategically important locales are listed and communicated to the engineering team.

**Comment by atopal** [comment]
We need about 12 languages to get to 99% coverage. Can you outline the issue with pre-rendering for more than 10 locales and our options? One concern I have is the UX for 1% of MDN users, another concern is the hard to predict nature of Google rating the performance of our domain. If they use the googlebot for that, we might take a hit. It might be worth it, but I'd like to understand why we are making that trade off.

Here are the 12 locales: 

- en-US
- fr
- es
- zh-CN
- ru
- ja
- pt-BR
- de
- ko
- zh-TW
- pl
- it

**Comment by davidflanagan** [comment]
@atopal Thanks for the list of locales.  (cc @escattone)

Here's the context for this user story:

Whenever a writer edits a page, we currently re-render the JSON file that is used to render that page for client-side navigation. But we would also like to re-server-side-render the HTML file that is used for a regular page load for that document.

Suppose the document is an english document that has not been translated into any other languages. In that case we'll be displaying the english document in every locale. If we've got 60 locales, that means there could be 60 different HTML files that contain the same document and have locale-specific strings for the menus and other page UI.

According to your stats, 48 of those 60 locales add up to less than 1% of our traffic. And  most of those are presumably less than 0.01% each. Given that, I don't think it makes sense to pre-render to those 48 locales on every edit of the document. Instead, we can just let those locales do the SSR on demand (which is what we are doing for all locales today.)

Note that the 48 minor locales presumably have almost no translations, so basically any edit to any english document on MDN would trigger a re-render in all of those locales. It would increase our server load and our S3 storage bill for almost no benefit.

What we're proposing here should be a nice performance boost for the most important locales, but we do not expect it to regress performance for the minor locales.

Does that make sense? And does the list of locales you posted above still seem right given this context?  If so, then we can mark this user story complete before even pulling it into a sprint.

At a gut level, 12 seems like the maximum number of locales that I'd want to be pre-rendering on every edit. Is there a reasonable way we can get it down a bit? What if we used 95% as the threshold instead of 99%? How would that affect the set of locales? Or what if we just took the locales that are themselves more than 1% or 2% of traffic?
